File: app/src/main/java/com/nobrain/samples/daumeimagesearch/home/adapter/ImageAdapterDataModel.java
package com.nobrain.samples.daumeimagesearch.home.adapter;

import com.nobrain.samples.daumeimagesearch.domain.search.ImageItem;

public interface ImageAdapterDataModel {
    void add(ImageItem imageItem);
    int getSize();
    ImageItem getItem(int position);
    void clear();
}

File: app/src/main/java/com/nobrain/samples/daumeimagesearch/home/adapter/ImageAdapterDataModel.kt
package com.nobrain.samples.daumeimagesearch.home.adapter

import com.nobrain.samples.daumeimagesearch.domain.search.ImageItem

interface ImageAdapterDataModel {
    fun add(imageItem: ImageItem)
    val size: Int
    fun getItem(position: Int): ImageItem
    fun clear()
}
